### PR TITLE
Add Spring and Vue blog post

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 1]
```diff
@@ -0,0 +1,1133 @@
...
+  },  
+  
+  handleErrorClick: function () {  
+    this.error = null;  
+  },  
+  
+},
+```
+
+Notice that the methods defined in the `methods()` block do not use the arrow syntax, `=>`. These functions will be bound to the appropriate context by Vue later. Using the `=>` here will not work because the module has not been created yet, so `this` will refer to the `window`, which isn't what we want. Potentially confusingly, however, notice that the api callback methods *do* use the arrow syntax. When the module's functions are bound to the `this` instance of the module, these arrow functions allow the callbacks to refer to the module's context.
```
Comment (mraible) [inline _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: api > API

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 2]
```diff
@@ -0,0 +1,1133 @@
...
+
+The flexibility of binding `this` in JavaScript is both one of its strengths and one of its most confusing aspects. 
+
+Bask in the glory of your todo app! We still need to add security, but otherwise we have a fulling functioning todo app that's creating, reading, updating, and deleting data on the server. Sweet.
```
Comment (mraible) [inline _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: We > You

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 3]
```diff
@@ -0,0 +1,1133 @@
...
+
+This point of the tutorial corresponds to the **pre-auth** branch of the repository.
+
+Go ahead and run it using `yarn serve`. Make sure your Spring Boot resource server is still running.
+
+## Integrate Okta and Add User Authentication
+
+Okta makes user authentication really easy. The first step is to sign up for a free [
```
Comment (mraible) [inline _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: missing caption for link

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 4]
```diff
@@ -0,0 +1,1133 @@
...
+  el: '#app',  
+  router,  // <-- add this line
+  template: '<App/>',  
+  components: { App }  
+})
+```
+
+Next, update the `src/App.vue` module to match the following:
+
+```js
```
Comment (mraible) [inline _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: html
```suggestion
```html
```

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 5]
```diff
@@ -0,0 +1,1133 @@
...
+
+{% img blog/build-crud-spring-and-vue/login-prompt.png alt:"Okta Login screenshot" width:"600" %}{: .center-image }
+
+{% img blog/build-crud-spring-and-vue/todo-app-screenshot-auth.png alt:"" width:"800" %}{: .center-http :9000/todosimage }
+
+There's still one more step to go, however. You can log in through Okta, but the Spring Boot server app isn't requiring authentication. 
+
+## Configure Spring Boot Server For Token Auth
+
+Okta has a nice project call the Okta Spring Boot Starter ([check out the Github page](https://github.com/okta/okta-spring-boot)) that makes added token authentication to a Spring Boot project quick and painless.
```
Comment (mraible) [inline _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: Github > GitHub

Comment (mraible) [reply _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: page > project

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 6]
```diff
@@ -0,0 +1,1133 @@
...
+  oauth2:  
+    issuer: https://{yourOktaDomain}/oauth2/default  
+    clientId: {yourClientId}  
+    scope: openid profile email
+```
+
+Finally, you need to add the `@EnableResourceServer` annotation to your `SpringBootVueApplication` class.
+
+```java
+@EnableResourceServer  // <- add me
```
Comment (mraible) [inline _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: Add import for those that don't use IDEs

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 7]
```diff
@@ -0,0 +1,1133 @@
...
+   }
+   ...
+}
+```
+
+And that's it! You now have a fully functioning Vue client and Spring Boot rest service using Okta's authentication services. Too easy.
+
+## Do More with Okta, Vue, and Spring Boot
+
+This tutorial did quite a lot. you built a Vue.js client application and a Spring Boot rest service, using them to demonstrate a fully functioning CRUD application. You also added authentication using Okta and the Okta Vue SDK.
```
Comment (mraible) [inline _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: you > You

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 8]
```diff
@@ -0,0 +1,1133 @@
...
+}
+```
+
+And that's it! You now have a fully functioning Vue client and Spring Boot rest service using Okta's authentication services. Too easy.
+
+## Do More with Okta, Vue, and Spring Boot
+
+This tutorial did quite a lot. you built a Vue.js client application and a Spring Boot rest service, using them to demonstrate a fully functioning CRUD application. You also added authentication using Okta and the Okta Vue SDK. 
+
+If you'd like to dig a little deeper, take a look at [the Okta Vue SDK project page](https://github.com/okta/okta-oidc-js/tree/master/packages/okta-vue).
```
Comment (mraible) [inline _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: delete page

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 9]
```diff
@@ -0,0 +1,1133 @@
...
+
+The Spring Boot rest service used Spring Data's JPA implementation to persist data based on a Java class. Spring Data and JPA is a super deep area, and [the Spring docs on it](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/) are a great place to learn more.
```
Comment (mraible) [inline _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: rest > REST

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 10]
```diff
@@ -0,0 +1,1133 @@
...
+
+Okta also has a number of other great related tutorials.
+[Build a Basic CRUD App with Angular 5.0 and Spring Boot 2.0](/blog/2017/12/04/basic-crud-angular-and-spring-boot)
```
Comment (mraible) [inline _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: These need asterisks in front of them

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md (hunk 6 above)
Comment (bdemers) [reply _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: added
```
import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer;
...
```
what do you think?

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 11]
```diff
@@ -0,0 +1,1133 @@
...
+<style>
+  [v-cloak] { display: none; }
+</style>
+```
+
+Delete the `src/components/HelloWorld.vue` module. You can also delete the `src/assets` folder, if you want, as we won't need it.
+
+Create a new Vue component called `src/components/Todos.vue`:
+
+```js
```
[the remark this answers is not in the document]
Comment (bdemers) [reply _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: Why raw? it seems to render fine with `html` ?

File: _source/_posts/2018-11-19-build-crud-spring-and-vue.md [hunk 12]
```diff
@@ -0,0 +1,1133 @@
...
+- "Tutorial - @SpringBoot & @VueJS example application"
+- "Client/Server application tutorial with #SpringBoot and #VuewJS"
+image: blog/featured/okta-java-short-headphones.jpg
+---
+
+In this tutorial, you're going to build a complete CRUD web application using Vue.js for the client and Spring Boot as the resource server. You'll also secure the application using OAuth 2.0 and Okta. 
+
+CRUD is **C**reate, **R**ead, **U**pdate, and **D**elete. It's kinda the "Hello World" of the server world. Like "Hello server!" If you can add data, updated it, read it, and delete it, you've pretty much got all the basic tools down for a REST interface or basic resource API.
+
+The example application you're going to build is a simple todo app. Typically these todo apps use data stored locally, but in this example you're going to create, read, update, and delete todos on a Spring Boot resource server.
```
[the remark this answers is not in the document]
Comment (bdemers) [reply _source/_posts/2018-11-19-build-crud-spring-and-vue.md]: I tweaked a couple points, but it looks like Todo is capitalized below when referencing classes or components?

**Comment by bdemers** [comment]
@mraible back to you

**Comment by mraible** [comment]
Looks good!

**Pull request merged**